packages/xod-client/src/project/actions.js

```diff
@@ -65,6 +72,50 @@ export const openWorkspace = libs => ({
   payload: libs,
 });
 
+export const requestPublishProject = () => (dispatch, getState) => {
+  const isAuthorised = foldMaybe(false, R.T, getGrant(getState()));
+
+  if (!isAuthorised) {
+    dispatch(addError(LOG_IN_TO_CONTINUE));
+    // TODO: open account pane?
+    return;
+  }
+
+  dispatch({ type: ActionType.PROJECT_PUBLISH_REQUESTED });
+};
+
+export const cancelPublishingProject = () => ({
+  type: ActionType.PROJECT_PUBLISH_CANCELLED,
+});
+
+export const publishProject = () => (dispatch, getState) => {
+  const state = getState();
+  const project = getProject(state);
+
+  dispatch({ type: ActionType.PROJECT_PUBLISH_START });
+
+  dispatch(fetchGrant()) // to obtain freshest auth token
+    .then((freshGrant) => {
+      if (freshGrant == null) {
+        // could happen if user logs out in another tab
+        dispatch(addError(LOG_IN_TO_CONTINUE));
```

Comment:
DRY?
Move `addError` dispatching (and possibly showing of account pane) into another function and call it something like `requestLogIn`?